Raven.Database/Actions/DocumentActions.cs

```diff
@@ -273,6 +273,8 @@ public int BulkInsert(BulkInsertOptions options, IEnumerable<IEnumerable<JsonDoc
                     cts.Token.ThrowIfCancellationRequested();
 
                     var docsToInsert = docs.ToArray();
+                    docsToInsert.ForEach(document => document.Key = string.IsNullOrWhiteSpace(document.Key) ? Guid.NewGuid().ToString() : document.Key.Trim());
```

Comment:
I think you are right, we should move the code to the next loop.
For the new Guid we work like this in the session.Store we put
a new Guid if the key is null. For the consistency I think this should stay and I will delete the if statement